### DISCUSSION TITLE
Document in mail missing when sending through messenger

### DISCUSSION THREAD

**Discussion by valentin-** [pimcore/pimcore#16137] [Q&A / Support] [answered]
Hi, 

I'm currently facing a weird behaviour when sending mails (Pimcore 10 and 11). Im not sure if it's related to the ecommerce bundle. When finishing an order a symfony message is created to send the mail. After 2-3 orders the document variable (email document) is not available in the email.html.twig template.

After restarting the worker (console messenger:consume) it's working fine again (for a few orders).

When sending the mails directly using a command there is no problem, only when sent using the messenger.

For debug purpose I've set the document id of the mail document. The ID gets correctly fetched from the object, so it's definetly there.

![image](https://github.com/pimcore/pimcore/assets/90033/7048edad-2275-4660-b829-cee9c68084ca)

Can anyone imagine why the document variable isn't always available?

![image](https://github.com/pimcore/pimcore/assets/90033/f6a3f503-9c28-4b57-a481-5ac6e2d4173e)

Thanks in advance,
Valentin

**Comment by dvesh3** [comment, marked as answer]
Hi @valentin-, this seems related to https://github.com/pimcore/pimcore/pull/15908. Could you please check if the issue is resolved already on latest version v11.1 or v11.0.12? thanks!

**Reply by valentin-** [reply]
Thanks for the info. The error wasn't alway reproducable but didn't occur so far after the update to 11.0.12.
